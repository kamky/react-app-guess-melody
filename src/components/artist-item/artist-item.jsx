// React
import React from "react";
import PropTypes from "prop-types";

// Component ArtistItem
export const ArtistItem = (props) => {
  const {index, answer, onUserAnswer} = props;

  const handleInputChange = () => {
    onUserAnswer(answer);
  };

  return (
    <div className="artist">
      <input
        className="artist__input visually-hidden"
        type="radio"
        name="answer"
        value={`answer-${index}`}
        id={`answer-${index}`}
        onChange={handleInputChange}
      />
      <label className="artist__name" htmlFor={`answer-${index}`}>
        <img
          className="artist__picture"
          src={answer.picture}
          alt={answer.artist}
        />
        {answer.artist}
      </label>
    </div>
  );
};

ArtistItem.propTypes = {
  index: PropTypes.number.isRequired,
  onUserAnswer: PropTypes.func.isRequired,
  answer: PropTypes.shape({
    artist: PropTypes.string.isRequired,
    picture: PropTypes.string.isRequired,
  }).isRequired,
};
