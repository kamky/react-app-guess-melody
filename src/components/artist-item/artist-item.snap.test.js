// React
import React from "react";
import renderer from "react-test-renderer";

// Components
import {ArtistItem} from "./artist-item";

describe(`Snapshot test for component <ArtistItem />`, () => {
  const defaultProps = {
    index: 0,
    onUserAnswer: jest.fn(),
    answer: {artist: `artist-name`, picture: `picture-url`},
  };
  const wrapper = renderer.create(<ArtistItem {...defaultProps} />);
  test(`should component <ArtistItem /> renders correctly`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
