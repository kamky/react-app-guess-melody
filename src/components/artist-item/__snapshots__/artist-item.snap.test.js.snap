// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for component <ArtistItem /> should component <ArtistItem /> renders correctly 1`] = `
<div
  className="artist"
>
  <input
    className="artist__input visually-hidden"
    id="answer-0"
    name="answer"
    onChange={[Function]}
    type="radio"
    value="answer-0"
  />
  <label
    className="artist__name"
    htmlFor="answer-0"
  >
    <img
      alt="artist-name"
      className="artist__picture"
      src="picture-url"
    />
    artist-name
  </label>
</div>
`;
